Computer.cpp

```diff
@@ -0,0 +1,124 @@
+#include<iostream>
+#include"Computer.h"
+
+Computer::Computer()
+{
+    std::cout << "Computer being constructed!" << std::endl;
+}
+
+Computer::~Computer()
+{
+    std::cout << "Computer being destructed!" << std::endl;
+}
+
+Computer::Hardware::Hardware()
+{
+    std::cout << "Computer Hardware being constructed!\n" << std::endl;
+}
+
+Computer::Hardware::~Hardware()
+{
+    std::cout << "Computer Hardware being destructed!\n" << std::endl;
+}
+
+std::string Computer::Hardware::playGames()
+{
+    if (RAM >= 4 && disk >=256 && CPU >= 3.0 && GPU >= 2)
+    {
+        canPlayGames = true;
+    }
+    return "You can use this computer to play games!"; 
+}
+
+void Computer::Hardware::trainAI()
+{
+    if (canPlayGames == true && GPU > 4)
+    {
+        //std::cout << "You can use this computer to train AI using" << softwareReq;
+        canTrainAI = true;
+    }
+    
+}
+
+void Computer::Hardware::workAtOffice()
+{
+    if (canPlayGames == true)
+    {
+        //std::cout << "You can use this computer to do " << mostlyUsedTask;
+        canUseAtOffice = true;
+    }
+    
+} 
+
+bool Computer::executePrograms(Hardware* specs, std::string installedSoft)
+{
+    if (installedSoft == "GTA")
+    {
+        if (specs->canPlayGames == true)
+        {
+            std::cout << "According to the given specs, this computer can be used to play Grand Theft Auto\n" << std::endl;
+        }
+        else
+        {
+            std::cout << "Sorry!! This computer can NOT be used to play Grand Theft Auto becasuse of the specs" << std::endl;
+        }
+    }
+    return specs->canPlayGames;
+}
+
+std::string Computer::saveInfo(bool diskAvailable)
+{
+    if (diskAvailable == true)
+    {
+        canSaveInfo = true;
+    }
+    return "The file was saved";
+    
+}
+
+void Computer::connectToPCs(bool LANavailable)
+{
+    if (LANavailable == true)
+    {
+        std::cout << "The computer is available to connect to LAN";
+    }
+}
+
+Computer::Hardware Computer::addGamesToRAM(Hardware* pcType, bool abilityToPlay)
+{
+    if (abilityToPlay == true)
```

Comment:
If you can, pass by reference or const reference.
Only pass pointers when you aren't sure of the lifetime of the object being passed in.
- [ ] revise